File: example-android-project/app/src/main/java/io/github/gofaith/faithdroid/FViews/FEditText.java
package io.github.gofaith.faithdroid.FViews;

import android.graphics.Color;
import android.support.v4.view.ViewCompat;
import android.text.InputType;
import android.view.View;
import android.widget.EditText;

import faithdroid.Faithdroid;
import io.github.gofaith.faithdroid.UI.AttrGettable;
import io.github.gofaith.faithdroid.UI.AttrSettable;
import io.github.gofaith.faithdroid.UI.UIController;

public class FEditText extends FView implements AttrGettable, AttrSettable {
    public EditText v;

    public FEditText(UIController controller) {
        parentController=controller;
        v = new EditText(parentController.activity);
        v.setId(ViewCompat.generateViewId());
        view=v;
    }
    @Override
    public String getAttr(String attr) {
        String str = getUniversalAttr(attr);
        if (str != null) {
            return str;
        }
        switch (attr) {
            // ------------------------------------------
            case "Text":
                return v.getText().toString();
        }
        return "";
    }

    @Override
    public void setAttr(String attr, final String value) {
        if (setUniversalAttr(attr, value)) {
            return;
        }
        switch (attr) {
            // -------------------------------------------------------------------
            case "Text":
                v.setText(value);
                break;
            case "TextColor":
                try {
                    v.setTextColor(Color.parseColor(value));
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;
            case "TextSize":
                try {
                    v.setTextSize(dp2pixel(parentController.activity, Float.valueOf(value)));
                } catch (Exception e) {
                    e.printStackTrace();
                    return;
                }
                break;
            case "InputType":
                int it;
                if (value.equals("Text")) {
                    it=InputType.TYPE_CLASS_TEXT;
                } else if (value.equals("Number")) {
                    it = InputType.TYPE_CLASS_NUMBER;
                } else if (value.equals("Password")) {
                    it=InputType.TYPE_CLASS_TEXT|InputType.TYPE_TEXT_VARIATION_WEB_PASSWORD;
                } else  {
                    it = InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD;
                }
                v.setInputType(it);
                break;
        }
    }
}

File: android/app/src/main/java/io/github/stevenzack/faithdroidapp/FViews/FEditText.kt
package io.github.stevenzack.faithdroidapp.FViews

import android.graphics.Color
import android.text.InputFilter
import android.text.InputType
import android.widget.EditText
import io.github.stevenzack.faithdroidapp.UI.AttrGettable
import io.github.stevenzack.faithdroidapp.UI.AttrSettable
import io.github.stevenzack.faithdroidapp.UI.FView
import io.github.stevenzack.faithdroidapp.UI.Toolkit.dp2pixel
import io.github.stevenzack.faithdroidapp.UI.UIController

class FEditText(controller: UIController) : FView(), AttrGettable, AttrSettable {
    var v: EditText

    init {
        parentController = controller
        v = EditText(parentController!!.activity)
        view = v
    }

   override fun getAttr(attr: String): String {
        val str = getUniversalAttr(attr)
        if (str != null) {
            return str
        }
        when (attr) {
            // ------------------------------------------
            "Text" -> return v.getText().toString()
        }
        return ""
    }

   override fun setAttr(attr: String, value: String) {
        if (setUniversalAttr(attr, value)) {
            return
        }
        when (attr) {
            // -------------------------------------------------------------------
            "Text" -> v.setText(value)
            "TextColor" -> try {
                v.setTextColor(Color.parseColor(value))
            } catch (e: Exception) {
                e.printStackTrace()
            }

            "TextSize" -> try {
                v.setTextSize(dp2pixel(parentController!!.activity, java.lang.Float.valueOf(value)))
            } catch (e: Exception) {
                e.printStackTrace()
                return
            }

            "InputType" -> {
                val it: Int
                if (value == "Text") {
                    it = InputType.TYPE_CLASS_TEXT
                } else if (value == "Number") {
                    it = InputType.TYPE_CLASS_NUMBER
                } else if (value == "Password") {
                    it = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_WEB_PASSWORD
                } else {
                    it = InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD
                }
                v.setInputType(it)
            }
            "MaxLines" -> v.setMaxLines(Integer.parseInt(value))
            "MaxEms" -> v.setMaxEms(Integer.parseInt(value))
            "Hint" -> v.setHint(value)
            "MaxLength" -> v.setFilters(arrayOf<InputFilter>(InputFilter.LengthFilter(Integer.parseInt(value))))
        }
    }
}
